src/commands/publish/upload_form/mod.rs

```diff
@@ -0,0 +1,139 @@
+mod file;
+mod project_assets;
+mod wasm_module;
+
+use log::info;
+
+use reqwest::multipart::{Form, Part};
+use std::fs;
+use std::path::Path;
+
+use crate::commands::build::wranglerjs::Bundle;
+use crate::settings::binding;
+use crate::settings::metadata::Metadata;
+use crate::settings::project::{Project, ProjectType};
+
+use project_assets::ProjectAssets;
+use wasm_module::WasmModule;
+
+use super::{krate, Package};
+
+pub fn build_script_upload_form(project: &Project) -> Result<Form, failure::Error> {
+    let project_type = &project.project_type;
+    match project_type {
+        ProjectType::Rust => {
+            info!("Rust project detected. Publishing...");
+            let name = krate::Krate::new("./")?.name.replace("-", "_");
+            // TODO: move into build?
+            build_generated_dir()?;
+            concat_js(&name)?;
+
+            let wasm_module = WasmModule {
+                path: format!("./pkg/{}_bg.wasm", name).to_string(),
+                filename: "wasmprogram".to_string(),
+                binding: "wasm".to_string(),
+            };
+
+            let script_path = "./worker/generated/script.js".to_string();
+
+            let assets = ProjectAssets {
+                script_path,
+                wasm_modules: vec![wasm_module],
+            };
+
+            build_form(&assets)
+        }
+        ProjectType::JavaScript => {
+            info!("JavaScript project detected. Publishing...");
+            let package = Package::new("./")?;
+
+            let script_path = package.main()?;
+
+            let assets = ProjectAssets {
+                script_path,
+                wasm_modules: Vec::new(),
+            };
+
+            build_form(&assets)
+        }
+        ProjectType::Webpack => {
+            info!("Webpack project detected. Publishing...");
+            // FIXME(sven): shouldn't new
+            let bundle = Bundle::new();
+
+            let script_path = bundle.script_path();
+
+            let mut wasm_modules = Vec::new();
+
+            if bundle.has_wasm() {
+                let wasm_module = WasmModule {
+                    path: bundle.wasm_path(),
+                    filename: bundle.get_wasm_binding(),
```

Comment:
so let's make filename/binding the same in Webpack as RustWasm and always point to the same thing - either wasm or wasmprogram - we can file another issue for it

Reply:
https://github.com/cloudflare/wrangler/issues/358